### DISCUSSION TITLE
Virtual Bids

### DISCUSSION THREAD

**Discussion by JeffDecary** [ANL-CEEESA/UnitCommitment.jl#22] [Feature Requests] [answered]
Hi,

I am getting started with this package, and I have a few questions for the authors. 

First, I am trying to modify some of the constraints, but all variables are reported as Noname when I print the model. I am playing with the source code, but I wonder if I am missing something. 

Second, I am hoping to explore virtual bids. I read that you are planning to add them. Are there already instances with virtual bids? If not, do you have advice on the easiest way to implement them?

Thank you,
Jeff

**Comment by iSoron** [comment, marked as answer]
Hi Jeff, thank you for your interest in the package.

1. We don't set variable names by default for performance reasons. If you need them for debugging, you can call `build_model` as follows:
    ```julia
    model = UnitCommitment.build_model(
        instance = instance,
        optimizer = Cbc.Optimizer,
        variable_names = true,
    )
    ```
2. Our data format does not currently mention virtual transactions, and the package does not currently have any specialized algorithms to handle them more efficiently, but it is already possible to model them. For *virtual demand bids*, you can enter them as price-sensitive loads, since these two concepts are mathematically equivalent. For *virtual supply bids*, it is possible to enter them as regular generators. Note that, by default, generators have no ramping/startup/shutdown restrictions, so you just need to provide the virtual offer curve. Implementing virtuals more efficiently would be a nice contribution to the package, if anyone is interesting in collaborating. Our instances also do not currently have any virtuals. Generating them realistically would also be a nice contribution.
